### PR TITLE
#140227231 Authenticate Endpoints

### PR DESCRIPTION
**Pull request by jackiewmacharia**
#### What does this PR do?
Authenticates endpoints created in [pull 3](https://github.com/andela/arakari-rc/pull/3)

 #### How should this be manually tested?
###### Using Postman:
* Make a login post request on Postman and login in an existing user by adding the email and password in body:

>http://localhost:9090/api/login/

* Set the token returned as a value to the x-access-token key in headers and make the following get requests on Postman:
>http://localhost:9090/api/products/
>http://localhost:9090/api/users/
>http://localhost:9090/api/shops/
>http://localhost:9090/api/ordered_items/admin
>http://localhost:9090/api/orders_processed/admin
>http://localhost:9090/api/orders_cancelled/admin

 #### What are the relevant pivotal tracker stories?
#140227231

 #### Screenshots (if appropriate)
screen shot 2017-03-21 at 08 07 16

screen shot 2017-03-20 at 14 46 20
screen shot 2017-03-20 at 14 46 27

### CODE REVIEW

File: server/api/graphql/graphqlServer.js [hunk 1]
```diff
@@ -13,6 +20,45 @@ app.use("/graphql", GraphQLHTTP({
...
+  }
+  const password = request.body.password;
+  const hash = user[0].services.password.bcrypt;
+  return bcrypt.compare(SHA256(password), hash, (err, res) => {
+    if (err) {
+      return response.status(500).send(err);
+    } else if (!res) {
+      return response.status(401).send({message: "Authentication failed"});
+    }
+    const token = jwt.sign(user[0], secret, {expiresIn: "24h"});
```
Comment (boswellgathu) [inline server/api/graphql/graphqlServer.js]: when using the whole user object, you make the token created to be a really long string, plus a lot of user data is exposed, maybe you can choose what should be in the user object before using it to create the token

Comment (boswellgathu) [reply server/api/graphql/graphqlServer.js]: so what you can do is either use the user.id only or any other user information that is used while authenticating, but userId is usually enough

**Review by boswellgathu — CHANGES_REQUESTED** [review]

File: server/api/graphql/graphqlServer.js (hunk 1 above)
Comment (jackiewmacharia) [reply server/api/graphql/graphqlServer.js]: @andela-Bgathu On it

Comment (jackiewmacharia) [reply server/api/graphql/graphqlServer.js]: @andela-Bgathu done

**Review by boswellgathu — APPROVED** [review]

**Pull request merged**